dmr/src/main/java/org/jboss/hal/dmr/dispatch/Dispatcher.java

```diff
@@ -173,94 +177,70 @@ public Dispatcher(Environment environment, Endpoints endpoints, Settings setting
     // ------------------------------------------------------ execute composite
 
     @JsIgnore
-    public void execute(Composite composite, CompositeCallback callback) {
-        dmr(composite, payload -> payload.get(RESULT), callback, failedCallback, exceptionCallback);
+    public void execute(Composite operations, CompositeCallback success) {
+        dmr(operations, res -> success.onSuccess(payload(res)), failedCallback, exceptionCallback);
     }
 
     @JsIgnore
-    public void execute(Composite composite, CompositeCallback callback, FailedCallback failedCallback) {
-        dmr(composite, payload -> payload.get(RESULT), callback, failedCallback, exceptionCallback);
+    public void execute(Composite operations, CompositeCallback success, OnFail fail) {
+        dmr(operations, res -> success.onSuccess(payload(res)), fail, exceptionCallback);
     }
 
     @JsIgnore
-    public void execute(Composite composite, CompositeCallback callback, FailedCallback failedCallback,
-            ExceptionCallback exceptionCallback) {
-        dmr(composite, payload -> payload.get(RESULT), callback, failedCallback, exceptionCallback);
+    public void execute(Composite operations, CompositeCallback success, OnFail fail, OnError error) {
+        dmr(operations, res -> success.onSuccess(payload(res)), fail, error);
     }
 
 
     // ------------------------------------------------------ execute composite in flow
 
     @JsIgnore
-    public void executeInFlow(Control control, Composite composite, CompositeCallback callback) {
-        dmr(composite, payload -> payload.get(RESULT), callback, new FailedFlowCallback<>(control),
-                new ExceptionalFlowCallback<>(control));
+    public void executeInFlow(Control failOrError, Composite operations, CompositeCallback success) {
```

Comment:
What's the intention behind renaming the `Control` parameter from `control` &rarr; `failOnError`?

Reply:
to emphasize that it is only used to handle the fail or the error cases, never used as a success. I made a refactor to move it to the right, but it changes a lot of code and makes the PR much less readable. But, if you see how it is used, the control is just an "obscure" combination of the fail and error callback and it ignores (never called) the success, at least by the dispatcher.

Reply:
The control is also used to proceed to the next step. So I'm rather going to keep the current naming.